File: Assignment7/Assignment7.py
# Name: Assignment7.py
# Description: This script answers basic python syntax and operations from Assignment 7.
# All work is completed by Judah Alter.

# 1. Plot log(x) (red) and sqrt(x) (green) on [1,5]
import numpy as np
import matplotlib.pyplot as plt
x = np.linspace(1, 5, 100)
plt.plot(x, np.log(x), color='red')
plt.plot(x, np.sqrt(x), color='green')
plt.xlabel('x')
plt.ylabel('f1')
plt.show()

# 2. Scatterplot of x and y, blue mean point, red lines at y=5 and y=15
x = [5,6,7,5,10,12,14]
y = [4,9,2,3,4,5,20]
plt.scatter(x, y)
plt.scatter(np.mean(x), np.mean(y), color='blue')
plt.axhline(5, color='red')
plt.axhline(15, color='red')
plt.xlabel('x')
plt.ylabel('y')
plt.show()

# 3a. Relative frequency histogram of Life Expectancy 
import pandas as pd
df = pd.read_csv('Datasets:UseCase/statesx77.txt', delim_whitespace=True)
plt.hist(df['Life_Exp'], density=True)
plt.xlabel('Life Expectancy')
plt.ylabel('Density')
plt.title('Histogram of Life Expectancy')
plt.show()

# 3b. Boxplot of Life Expectancy from statesx77.txt
import pandas as pd
df = pd.read_csv('Datasets:UseCase/statesx77.txt', sep='\s+')
plt.boxplot(df['Life_Exp'])
plt.title('Boxplot of Life Expectancy')
plt.ylabel('Life Expectancy')
plt.show()

# 3c. Kernel density estimate of Life Expectancy from 
import seaborn as sns
sns.kdeplot(df['Life_Exp'])
plt.title('Kernel Density Estimate of Life Expectancy')
plt.xlabel('Life Expectancy')
plt.ylabel('Density')
plt.show()

# 4a. Side-by-side boxplots of dopamine level by status 
import pandas as pd
import seaborn as sns
schizo = pd.read_excel('Datasets:UseCase/schizo.xlsx')
sns.boxplot(x='Status', y='Dopamine', data=schizo, color='grey')
plt.title('Non-Psychotic vs Psychotic')
plt.xlabel('Status')
plt.ylabel('Dopamine')
plt.show()

# 4b. Side-by-side relative frequency histograms of dopamine level by status
import matplotlib.pyplot as plt
import pandas as pd

schizo = pd.read_excel('Datasets:UseCase/schizo.xlsx')
fig, axes = plt.subplots(1, 2, figsize=(10, 4))
axes[0].hist(schizo[schizo['Status'] == 'N']['Dopamine'], density=True, color='cyan')
axes[0].set_title('Histogram of Non-Psychotic Patients')
axes[0].set_xlabel('Dopamine')
axes[0].set_ylabel('Density')
axes[1].hist(schizo[schizo['Status'] == 'P']['Dopamine'], density=True, color='cyan')
axes[1].set_title('Histogram of Psychotic Patients')
axes[1].set_xlabel('Dopamine')
axes[1].set_ylabel('Density')
plt.tight_layout()
plt.show()

File: Assignment7/Assignment7.R
# Name: Assignment7.R
# Description: This script answers basic R syntax and operations from Assignment 7.
# All work is completed by Judah Alter.

# Color functions are really cool and weirldy useful lol.

# 1. Plot log(x) (red) and sqrt(x) (green) on [1,5]
x = seq(1, 5, length.out=100)
plot(x, log(x), type="l", col="red", ylab="f1", xlab="x")
lines(x, sqrt(x), col="green")

# 2. Scatterplot of x and y, blue mean point, red lines at y=5 and y=15
x = c(5,6,7,5,10,12,14)
y = c(4,9,2,3,4,5,20)
plot(x, y, xlab="x", ylab="y")
points(mean(x), mean(y), col="blue", pch=1)
abline(h=5, col="red")
abline(h=15, col="red")

# 3a. Relative frequency histogram of Life Expectancy 
df = as.data.frame(state.x77)
hist(df$"Life Exp", probability=TRUE, col="grey", xlab="Life Expectancy", main="Histogram of Life Expectancy")


# 3b. Boxplot of Life Expectancy
df = as.data.frame(state.x77)
boxplot(df$"Life Exp", col="grey", main="Boxplot of Life Expectancy", ylab="Life Expectancy")

# 3c. Kernel density estimate of Life Expectancy
df = as.data.frame(state.x77)
plot(density(df$"Life Exp"), main="Kernel Density Estimate of Life Expectancy", xlab="Life Expectancy")

# 4a. Side-by-side boxplots of dopamine level by status
library(readxl)
schizo = read_excel("Datasets:UseCase/schizo.xlsx")
boxplot(Dopamine ~ Status, data=schizo, col="grey", main="Non-psychotic vs Psychotic", xlab="Status", ylab="Dopamine")

# 4b. Side-by-side relative frequency histograms of dopamine level by status
par(mfrow=c(1,2))
hist(schizo$Dopamine[schizo$Status == "N"], probability=TRUE, col="cyan", main="N", xlab="Dopamine", ylab="Percent of Total")
hist(schizo$Dopamine[schizo$Status == "P"], probability=TRUE, col="cyan", main="P", xlab="Dopamine", ylab="Percent of Total")
par(mfrow=c(1,1))